plugins/global-search/src/utils/indexer/indexer.ts

```diff
@@ -0,0 +1,183 @@
+import {
+    type AnyNode,
+    Collection,
+    framer,
+    isComponentNode,
+    isFrameNode,
+    isTextNode,
+    isWebPageNode,
+} from "framer-plugin"
+import { TypedEventEmitter } from "./event-emitter"
+import { stripMarkup } from "./strip-markup"
+import { type IndexEntry, includedAttributes, type RootNode, shouldIndexNode } from "./types"
+
+async function getNodeName(node: AnyNode): Promise<string | null> {
+    if (isWebPageNode(node)) {
+        if (node.path === "/") return "Home"
+        if (!node.path) return "Web Page"
+        return node.path
+    }
+
+    if (isComponentNode(node)) {
+        return node.name ?? "Component"
+    }
+
+    if (isFrameNode(node)) {
+        return node.name ?? "Frame"
+    }
+
+    if (isTextNode(node)) {
+        return (await node.getText()) ?? null
+    }
+
+    return null
+}
+
+export class GlobalSearchIndexer {
+    private eventEmitter = new TypedEventEmitter()
```

Comment:
nitpick: `TypedEventEmitter` to me sounds like its a generic event emitter that supports types. Maybe rename `IndexerEventEmitter`?